client/src/components/SigninButton.svelte

```diff
@@ -1,19 +1,151 @@
 <script lang="ts">
   import Modal from "./Modal.svelte";
   import { authStore as user } from "$stores/auth";
+  import { loginWithGoogle, loginWithMicrosoft, loginWithEmail, onSignOut, createUser } from "$lib/firebase/auth";
 
   // check if the user is logged in with getAuth
   let openSignInModal = false;
+  let email = "";
+  let password = "";
+  let confirmPass = "";
+  let clickedButton = false;
+  let clickedButton2 = false;
+  let errorMessage = "";
+  async function createAccount() {
+    if (password == "") {
+      return;
+    }
+    if (password != confirmPass) {
+      errorMessage = "Password does not match";
+      password = "";
+      confirmPass = "";
+      return;
+    }
+    if (email == "") {
+      return;
+    } else {
```

Comment:
this `else` is redundant because the `if` returns

Reply:
resolved